metrics/pbsmetrics.go

```diff
@@ -0,0 +1,85 @@
+package metrics
+
+// The metrics module contains APIs which export metrics to some sort of aggregation service.
+// It should not import any other PBS modules, to avoid surprise circular dependencies.
+
+
+// PBSMetrics logs useful metrics to InfluxDB.
+//
+// Implementations of this interface should be threadsafe, so they can be used in multiple goroutines.
+type PBSMetrics interface {
+	// ServerStartedRequest should be called each time the /auction endpoint is hit.
+	StartAuctionRequest(requestInfo *AuctionRequestInfo) AuctionRequestFollowups
+
+	// BidderStartedRequest should be called just before PBS calls Adapter.Call
+	StartBidderRequest(auctionRequestInfo *AuctionRequestInfo, bidRequestInfo *BidRequestInfo) BidderRequestFollowups
+
+	// StartCookieSyncRequest should be called each time the /cookie_sync endpoint is hit.
+	StartCookieSyncRequest()
+}
+
+// RequestSource is the list of sources where requests might come from.
+// This obviously isn't comprehensive... just defined on an as-needed basis.
+type RequestSource int
+
+const (
+	// Safari has restrictive policies on 3rd party cookies. This helps measure how much of an effect that has.
+	SAFARI RequestSource = iota
+	APP
+	OTHER
+	UNKNOWN
```

Comment:
Let's add `desktop` and make it the default. Also Safari doesn't seem to fit here if the idea is platform.